spec/requests/odata/root_json_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe "root json" do
+  include_context "odata"
+
+  let(:path) { mission_api_route }
+
+  context "with no forms" do
+    it "renders as expected" do
+      expect_output({
+        "@odata.context": "http://www.example.com/en/m/#{get_mission.compact_name}/odata/v1/$metadata?mode=m",
```

Comment:
What is with the `mode=m` on the end of this? Would prefer to get that out of there since it's redundant information. I'm assuming it's due to the way the routes are setup. If you can find the code where this URL is built, we could try to figure out it.

Reply:
The engine's code:

```ruby
"@odata.context" => o_data_engine.metadata_url,
```

That seems to be calling a Rails helper. Snippet of the context when debugging the rails internal code at `route_set#define_url_helper`:

<img width="303" alt="Screen Shot 2020-04-24 at 17 35 45" src="https://user-images.githubusercontent.com/2047062/80266961-9ba65c00-8653-11ea-9a59-d171783a2028.png">

It seems Rails is keeping `mode` and none of the other context (e.g. `mission_name`), so perhaps that's some sort of special word...

If you have suggestions I'm all ears, but as a simple workaround I added a line to our hook to just delete that `?mode=m` portion of the URL.